### ISSUE TITLE
[url_launcher] Could not find gradle-4.2.0.jar

### ISSUE THREAD

**Issue by fvaldes0109** [flutter/flutter#130695]
### Is there an existing issue for this?

- [X] I have searched the [existing package issues](https://github.com/flutter/flutter/issues?q=is%3Aopen+is%3Aissue+label%3Apackage)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### What package does this bug report belong to?

url_launcher

### What target platforms are you seeing this bug on?

Android

### Have you already upgraded your packages?

Yes

### Steps to reproduce

After installing **url_launcher 6.1.12** I'm not able to build the project anymore.

**gradle --version**

```console
------------------------------------------------------------
Gradle 8.1.1
------------------------------------------------------------

Build time:   2023-04-21 12:31:26 UTC
Revision:     1cf537a851c635c364a4214885f8b9798051175b

Kotlin:       1.8.10
Groovy:       3.0.15
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          11.0.19 (Ubuntu 11.0.19+7-post-Ubuntu-0ubuntu120.04.1)
OS:           Linux 5.15.0-73-generic amd64

```

### Expected results

I already tried what's in the threads of #50001 and #22470 and still nothing. Also I'm able to manually go to the links in the output and download the .jar files.

### Actual results

Whenever I try to build, I get this output:

```
Launching lib/main.dart on SM A217M in debug mode...
main.dart:1

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':url_launcher_android'.
> Could not resolve all files for configuration ':url_launcher_android:classpath'.
   > Could not find gradle-4.2.0.jar (com.android.tools.build:gradle:4.2.0).
     Searched in the following locations:
         https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/4.2.0/gradle-4.2.0.jar
   > Could not find bundletool-1.1.0.jar (com.android.tools.build:bundletool:1.1.0).
     Searched in the following locations:
         https://dl.google.com/dl/android/maven2/com/android/tools/build/bundletool/1.1.0/bundletool-1.1.0.jar
> Failed to notify project evaluation listener.
   > Could not get unknown property 'android' for project ':url_launcher_android' of type org.gradle.api.Project.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

### Code sample

Dependencies in **android/app/build.gradle:**

```
dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'androidx.test:runner:1.1.1'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
}
```

**android/build.gradle:**

```
buildscript {
    ext.kotlin_version = '1.8.0'
    repositories {
        google()
        mavenCentral()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:7.3.0'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}

allprojects {
    repositories {
        google()
        mavenCentral()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

tasks.register("clean", Delete) {
    delete rootProject.buildDir
}
```

**android/gradle.properties:**

```
org.gradle.jvmargs=-Xmx1536M
android.useAndroidX=true
android.enableJetifier=true
```

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.5, on Ubuntu 20.04.6 LTS 5.15.0-73-generic,
    locale es_CU)
    • Flutter version 3.10.5 on channel stable at /home/fernando/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (hace 5 semanas), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /home/fernando/Android/Sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /home/fernando/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2021.3)
    • Android Studio at /home/fernando/android-studio
    • Flutter plugin version 71.2.3
    • Dart plugin version 213.7433
    • Java version OpenJDK Runtime Environment (build
      11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.80.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.56.0

[✓] VS Code (version 1.81.0-insider)
    • VS Code at /usr/share/code-insiders
    • Flutter extension version 3.66.0

[✓] Connected device (3 available)
    • SM A217M (mobile) • RF8R60NBZVF • android-arm64  • Android 12 (API 31)
    • Linux (desktop)   • linux       • linux-x64      • Ubuntu 20.04.6 LTS
      5.15.0-73-generic
    • Chrome (web)      • chrome      • web-javascript • Google Chrome
      114.0.5735.198

[✓] Network resources
    • All expected network resources are available.

```

</details>

**Comment by huycozy** [comment]
Hi @fvaldes0109 
Can you please share `android\gradle\wrapper\gradle-wrapper.properties` file content? And please also try running [plugin example](https://github.com/flutter/packages/tree/main/packages/url_launcher/url_launcher/example) to see if the issue still persists or not.

**Comment by fvaldes0109** [comment]
Here is my `android\gradle\wrapper\gradle-wrapper.properties`:

```
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip
```

I tried the plugin example and I got the same output as before:

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':url_launcher_android'.
> Could not resolve all files for configuration ':url_launcher_android:classpath'.
   > Could not find gradle-4.2.0.jar (com.android.tools.build:gradle:4.2.0).
     Searched in the following locations:
         https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/4.2.0/gradle-4.2.0.jar
   > Could not find bundletool-1.1.0.jar (com.android.tools.build:bundletool:1.1.0).
     Searched in the following locations:
         https://dl.google.com/dl/android/maven2/com/android/tools/build/bundletool/1.1.0/bundletool-1.1.0.jar
> Failed to notify project evaluation listener.
   > Could not get unknown property 'android' for project ':url_launcher_android' of type org.gradle.api.Project.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 15s
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

**Comment by fvaldes0109** [comment]
I managed to make a successful build on my project. I think the problem was due to my internet, since when it was building this time, my download speed stood stable between 300 and 500 kb/s (faster that what I usually have) and after a few minutes it finished compiled.
I guess I assumed that wasn't the problem since I was able to manually download the files. Anyway thanks a lot for your help and sorry for bothering.
